lib/slots.ts

```diff
@@ -39,8 +42,52 @@ export const NANOSECONDS = 'slot-nanoseconds';
 // Calendar
 export const CALENDAR_ID = 'slot-calendar-identifier';
 
+interface Slots {
+  // Instant
+  [EPOCHNANOSECONDS]: bigInt.BigInteger; // number? JSBI?
+
+  // TimeZone
+  [TIMEZONE_ID]: string;
+
+  // DateTime, Date, Time, YearMonth, MonthDay
+  [ISO_YEAR]: number;
+  [ISO_MONTH]: number;
+  [ISO_DAY]: number;
+  [ISO_HOUR]: number;
+  [ISO_MINUTE]: number;
+  [ISO_SECOND]: number;
+  [ISO_MILLISECOND]: number;
+  [ISO_MICROSECOND]: number;
+  [ISO_NANOSECOND]: number;
+  [CALENDAR]: Temporal.Calendar;
+
+  // Date, YearMonth, MonthDay common slots
+  [DATE_BRAND]: unknown;
```

Comment:
These should be boolean or `true`. Not sure which would be best, but probably `true`.

Reply:
IIRC the date/YM/MD brand check is currently implemented using only HasSlot — it doesn't matter what's in the slot, only that it is present.